dev/scaler/function.psm1

```diff
@@ -0,0 +1,208 @@
+## 
+## All scaler modules will accept the same parameters. The following parameters will be 
+## parsed from the storage Queue message sent by the Engine Function:
+##
+## - graphData: the result of the Azure Graph API query
+## - tagData: tag data, with set and saveState details
+## - direction: the direction to scale the resource (Up or Down)
+##
+#######################################################################################################################
+
+function Update-Resource {
+    param (
+        [Parameter(Mandatory = $true)]
+        [Object]
+        $graphData,
+
+        [Parameter(Mandatory = $true)]
+        [Hashtable]
+        $tagData,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $direction
+    )
+
+    # Set preference variables
+    $ErrorActionPreference = "Stop"
+
+#########################################################################################################################
+## 
+## RESOURCE SPECIFIC SCALE DATA
+##
+## Here you will need to gather the resource specific data you need to perform the scale operation.
+## The process involves taking the Graph API query result for a given resource and mapping which values
+## you need to manipulate to effectively scale... 
+## 
+## Using the below example for App Service Plans, we map available return values to their "Small", "Medium",
+## and "Large" equivalents. 
+##
+#######################################################################################################################
+
+    $workerSizeMap = @{
+        D1          = "Small"
+        D2          = "Medmium"
+        D3          = "Large"
+        Default     = "Small"
+        Medium      = "Medium"
+        Large       = "Large"
+        SmallV3     = "Small"
+        MediumV3    = "Medium"
+        LargeV3     = "Large"
+    }
+
+#######################################################################################################################
+##
+## RESOURCE BASE DATA
+## 
+## All resources will use the $baseData variable, consisting of the scale targets Name and 
+## Resource Group Name.  These values are needed as parameter values to execute the scaling
+## operation. 
+##
+## We also create an empty Hash Table "$config", which will be used to pass all required 
+## parameters to the scale command.
+##
+## Set the "$tags" object so that we can update with new tags after the scaling operation
+##
+#######################################################################################################################
+
+    $baseData = @{
+        ResourceGroupName = $graphData.resourceGroup
+        Name              = $graphData.name
+    }
+
+    $config = @{ }
+    $tags = $tagData.tags
+
+#######################################################################################################################
+##  
+## SCALING UP
+##
+## You will need to switch off of the direction sent from the Engine function.  Scaling up is the easier 
```

Comment:
I don't understand the first sentence.

Reply:
I assume you want to say: "Scaling up means bringing the resource back to its original state."

Reply:
I believe he is referencing the "Switch" statement in PowerShell. I agree the 'phrasiology' needs some work, haha.

Reply:
Ah yes, now it makes sense. Why not call it: The scaling starts with a switch-statement based on the scale direction (either up/down)